README.md

```diff
@@ -1 +1,47 @@
-# To-Do-App
\ No newline at end of file
+# To-Do List Application
+
+## Description
+
+This is a simple React-based To-Do List application that allows users to add, complete, and delete tasks. The tasks are stored in the browser's local storage, so they persist across page reloads.
+
+## Features
+
+- Add tasks with a simple input field.
+- Mark tasks as completed.
+- Delete tasks.
+- Persistent storage using browser's local storage.
+
+## Technologies Used
+
+- React
+- CSS (for styling)
+- Font Awesome (for the delete icon)
+
+## Installation
+
+1. **Clone the Repository**
+
+   ```bash
+   git clone https://github.com/Manahil-04/To-Do-App.git
+   cd To-Do-App
+   ```
+
+   Install Dependencies
+
+2. **Install the required dependencies using npm or yarn:**
+
+    ```bash
+    npm install
+    ```
+    or
+
+    ```bash
+    yarn install
+    ```
+
+3. **Start the Application**
+    Run the application locally:
+    ```bash
+    npm run dev
+    ```
+    The application will be accessible at http://localhost:5173 in your web browser. Note that the port number might vary if another service is already using port 3000. In such cases, the development server will suggest an available port, and you'll be able to access the application there (e.g., http://localhost:5174).
```

Comment:
A little correction in the README

"The application will be accessible at http://localhost:5173 in your web browser. Note that the port number might vary if another service is already using `port 3000`. In such cases, the development server will suggest an available port, and you'll be able to access the application there (e.g., http://localhost:5174)."

I guess you meant port 5173 instead of port 3000.